src/runtime/hexagon_remote/sim_host.cpp

```diff
@@ -0,0 +1,360 @@
+        const char *T = getenv("HL_HEXAGON_SIM_TRACE");
+        if (T && T[0] != 0) {
+            status = sim->SetTracing(HEX_TRACE_PC, T);
+            if (status != HEX_STAT_SUCCESS) {
+                printf("HexagonWrapper::SetTracing failed: %d\n", status);
+                return -1;
+            }
+        }
+    }
+
+    // Configure use of debugger
+    int pnum = 0;
+    char *s = getenv("HL_HEXAGON_SIM_DBG_PORT");
+    if (s && (pnum = atoi(s))) {
+        printf("Debugger port: %d\n", pnum);
+        status = sim->ConfigureRemoteDebug(pnum);
+        if (status != HEX_STAT_SUCCESS) {
+            printf("HexagonWrapper::ConfigureRemoteDebug failed: %d\n", status);
+            return -1;
+        }
+    }
+
+    status = sim->EndOfConfiguration();
+    if (status != HEX_STAT_SUCCESS) {
+        printf("HexagonWrapper::EndOfConfiguration failed: %d\n", status);
+        return -1;
+    }
+
+    status = sim->LoadExecutableBinary();
+    if (status != HEX_STAT_SUCCESS) {
+        printf("HexagonWrapper::LoadExecutableBinary failed: %d\n", status);
+        return -1;
+    }
+    return 0;
+}
+
+int write_memory(int dest, const void *src, int size) {
+    assert(sim);
+
+    while (size > 0) {
+        // WriteMemory only works with powers of 2, so align down. It
+        // also only writes up to 8 bytes, so we need to do this
+        // repeatedly until we've finished copying the buffer.
+        int next = 1;
+        if (size >= 8) next = 8;
+        else if (size >= 4) next = 4;
+        else if (size >= 2) next = 2;
+        HEXAPI_Status status = sim->WriteMemory(dest, next, *reinterpret_cast<const HEX_8u_t*>(src));
+        if (status != HEX_STAT_SUCCESS) {
+            printf("HexagonWrapper::WriteMemory failed: %d\n", status);
+            return -1;
+        }
+
+        size -= next;
+        dest += next;
+        src = reinterpret_cast<const char *>(src) + next;
+    }
+    return 0;
+}
+
+int read_memory(void *dest, int src, int size) {
+    assert(sim);
+
+    while (size > 0) {
+        // This is the same logic as in write_memory above.
+        int next = 1;
+        if (size >= 8) next = 8;
+        else if (size >= 4) next = 4;
+        else if (size >= 2) next = 2;
+        HEXAPI_Status status = sim->ReadMemory(src, next, dest);
+        if (status != HEX_STAT_SUCCESS) {
+            printf("HexagonWrapper::ReadMemory failed: %d\n", status);
+            return -1;
+        }
+
+        size -= next;
+        src += next;
+        dest = reinterpret_cast<char *>(dest) + next;
+    }
+    return 0;
+}
+
+int send_message(int msg, const std::vector<int> &arguments) {
+    assert(sim);
+
+    HEXAPI_Status status;
+
+    HEX_4u_t remote_msg = 0;
+    status = sim->ReadSymbolValue("rpc_call", &remote_msg);
+    if (status != HEX_STAT_SUCCESS) {
+        printf("HexagonWrapper::ReadSymbolValue(rpcmsg) failed: %d\n", status);
+        return -1;
+    }
+    if (0 != write_memory(remote_msg, &msg, 4)) { return -1; }
```

Comment:
Halide generally doesn't use the constant-on-LHS convention; please rewrite as call() != 0